lib/sync.js

```diff
@@ -82,7 +84,7 @@ module.exports = function resolveSync(x, options) {
     } else if (isCore(x)) {
         return x;
     } else {
-        var n = loadNodeModulesSync(x, absoluteStart);
+        var n = (conditions.length > 0 ? loadNodeModulesWithExportsSync : loadNodeModulesSync)(x, absoluteStart);
```

Comment:
i'm not actually sure this is the right condition to check here - in the future, if someone passes an empty array, it should resolve _nothing_ rather than _everything_.

I think the `ignoreExportsField` setting should determine which one is dispatched-to here.